build.gradle.kts

```diff
@@ -50,20 +50,21 @@ allprojects {
     }
 }
 
-tasks.register("clean", Delete::class) {
-    delete(rootProject.buildDir)
+val clean by tasks.registering(Delete::class) {
+    delete(rootProject.layout.buildDirectory)
 }
 
 /*
  * https://detekt.dev/docs/gettingstarted/git-pre-commit-hook
  * https://medium.com/@alistair.cerio/android-ktlint-and-pre-commit-git-hook-5dd606e230a9
  */
-tasks.register("installGitHook", Copy::class) {
-    description = "Adding git hook script to local working copy"
-    println("Installing git hook to ${rootProject.rootDir}/.git/hooks")
+val installGitHook by tasks.registering(Copy::class) {
+    description = "Adding Git hook script to local working copy"
     from(file("${rootProject.rootDir}/git_hooks/pre-commit"))
     into { file("${rootProject.rootDir}/.git/hooks") }
-    fileMode = 0x777
+    filePermissions {
+        unix("rwxr-xr-x")
+    }
```

Comment:
I tried to build and I got this issue. 
<img width="1474" alt="Capture d’écran 2024-08-30 à 00 02 33" src="https://github.com/user-attachments/assets/77ec7c8a-4832-4213-9caa-33826ba79c44">

the permission on my local pre-commit file was : `-r-xrwSrwt`

After updating manually the permission, no issue to build.

Reply:
That's an issue with the old `fileMode` option on recent Gradle versions, that's why it's deprecated, and will be removed in Gradle 9.0.
`fileMode` expects its value to use an octal number, but it's decimal by default. This is causing the wrong file permission. 
The new API proof avoids this issue.

Reply:
After some test. 
- Building a new repo from this branch no issue
- But if you already downloaded the repo before, you got the error `permission denied`

@MGaetan89 is checking to try to avoid this issue.

Reply:
It should be fixed with: https://github.com/SRGSSR/srgdataprovider-android/pull/45/commits/6f3ae2e80e09972682dbe274390e3d7f9c87924b
Now the target file is deleted first before copying the `pre-commit` hook. I was no longer having the permission issue on my side. Can you try again @Loic-Dumas?

I've also updated the `code_quality.yml` workflow in https://github.com/SRGSSR/srgdataprovider-android/pull/45/commits/58715028abccca7a906529e435af00b556ce9619, so it also runs on the `main` branch. This will let Gradle populate the GitHub Actions cache for future builds.

Reply:
Problem fixed. wp